packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts

```diff
@@ -201,47 +206,18 @@ export class FeatureDevController {
         }
     }
 
-    // TODO add type
-    private async processUserChatMessage(message: any) {
-        if (message.message === undefined) {
-            this.messenger.sendErrorMessage('chatMessage should be set', message.tabID, 0, undefined, undefined)
-            return
-        }
-
-        /**
-         * Don't attempt to process any chat messages when a workspace folder is not set.
-         * When the tab is first opened we will throw an error and lock the chat if the workspace
-         * folder is not found
-         */
-        const workspaceFolders = vscode.workspace.workspaceFolders
-        if (workspaceFolders === undefined || workspaceFolders.length === 0) {
-            return
-        }
-
-        let session
-        try {
-            getLogger().debug(`${featureName}: Processing message: ${message.message}`)
+    private processErrorChatMessage = (err: any, message: any, session: Session | undefined) => {
+        const skipTitle = true
+        const errorMessage = createUserFacingErrorMessage(
+            `${featureName} request failed: ${err.cause?.message ?? err.message}`
+        )
 
-            session = await this.sessionStorage.getSession(message.tabID)
+        let isSendErrorMessage = true
+        let defaultMessage = ''
+        const isDenyListedError = denyListedErrors.some(err => errorMessage.includes(err))
 
-            const authState = await AuthUtil.instance.getChatAuthState()
-            if (authState.amazonQ !== 'connected') {
-                await this.messenger.sendAuthNeededExceptionMessage(authState, message.tabID)
-                session.isAuthenticating = true
-                return
-            }
-
-            switch (session.state.phase) {
-                case 'Init':
-                case 'Approach':
-                    await this.onApproachGeneration(session, message.message, message.tabID)
-                    break
-                case 'Codegen':
-                    await this.onCodeGeneration(session, message.message, message.tabID)
-                    break
-            }
-        } catch (err: any) {
-            if (err instanceof ContentLengthError) {
+        switch (err.code) {
+            case ContentLengthError.name:
```

Comment:
Have we verified if the name property has correct Error name ?

Reply:
Yes, it was tested in some few different scenarios, all return the error name.